pkg/controller/cloneset/sync/cloneset_sync_utils.go

```diff
@@ -241,3 +240,26 @@ func isPodAvailable(coreControl clonesetcore.Control, pod *v1.Pod, minReadySecon
 	}
 	return coreControl.IsPodUpdateReady(pod, minReadySeconds)
 }
+
+func CalculatePartitionReplicas(partition *intstrutil.IntOrString, replicasPointer *int32) (int, error) {
```

Comment:
consider move the function to  a package that is common to all controllers, so that statefulset & daemonset can reuse this function